README.md

```diff
@@ -215,7 +215,20 @@ properties:
 
 > `function(selectedItem: any, stateAndHelpers: object)` | optional, no useful default
 
-Called when the user selects an item. Called with the item that was selected
+Called when the user selects an item and the selected item has changed. Called with the item that was selected
+and the new state of `downshift`. (see `onStateChange` for more info on
+`stateAndHelpers`).
+
+- `selectedItem`: The item that was just selected
+- `stateAndHelpers`: This is the same thing your `children` prop
+  function is called with (see [Child Callback Function](#child-callback-function))
+
+### onSelect
+
+> `function(selectedItem: any, stateAndHelpers: object)` | optional, no useful default
+
+Called when the user selects an item, regardless of the previous selected item. `onChange` is only called when the selected item is different from the previously selected one.
```

Comment:
Consider putting docs about `onChange` under `onChange`. I don't think it needs to be here in the `onSelect` section.

Reply:
I thought it could make sense since it might not be obvious what the difference between `onChange` and `onSelect` is at a glance. But I don't mind removing it again. (it's already documented in `onChange`

Reply:
I think that's a valid point. I thought things were clear enough without it. But that's why I said "consider" instead of "please do this", so you can do what seems best.